pkg/account/handler.go

```diff
@@ -0,0 +1,202 @@
+
+	return c.JSON(http.StatusOK, map[string]string{
+		"msg": "Successfully created"})
+
+}
+
+/**
+ * @todo cookie not implemented
+ */
+
+// @Summary login using email and passwd
+// @Description
+// @Tags Account
+// @Produce json
+// @Param email formData string true "user e-mail"
+// @Param paswd formData string true "user password"
+// @Success 200 {string} string "{"msg": "Successfully logged in"}"
+// @Failure 400 {string} string "{"error": "Invali E-mail Address"}"
+// @Router /case/PastHistory/{id} [PATCH]
+func (h *AccountHandler) LoginAccount(c echo.Context) error {
+	Email := c.QueryParam("Email")
+	Passwd := c.QueryParam("Passwd")
+
+	if ok, _ := regexp.MatchString(`^\w+@\w+[.\w+]+$`, Email); !ok {
+		return c.JSON(http.StatusBadRequest, map[string]string{
+			"error": "Invali E-mail Address"})
+	}
+
+	if len(Passwd) < ACCOUNT_PASSWD_LEN {
+		return c.JSON(http.StatusBadRequest, map[string]string{
+			"error": "Invali Password Length"})
+	}
+
+	return checkPasswd(c, Email, Passwd)
+}
+
+// @Summary reset password
+// @Description host will send a verification code to email, need response with verification code
+// @Tags Account
+// @Produce json
+// @Param email formData string true "user e-mail"
+// @param vericode formData string true "verification code sent by user"
+// @Param paswd formData string true "user password"
+// @Success 200 {string} string "{"msg": "Successfully modified"}"
+// @Failure 400 {string} string "{"error": "Invali E-mail Address"}"
+// @Router /case/PastHistory/{id} [PATCH]
+func (h *AccountHandler) ResetPasswd(c echo.Context) error {
+	Email := c.QueryParam("Email")
+
+	if ok, _ := regexp.MatchString(`^\w+@\w+[.\w+]+$`, Email); !ok {
+		return c.JSON(http.StatusBadRequest, map[string]string{
+			"error": "Invali E-mail Address"})
+	}
+
+	// Gen verification code
+	buffer := make([]byte, 6)
+	_, err := rand.Read(buffer)
+	if err != nil {
+		panic(err)
+	}
+	for i := 0; i < 6; i++ {
+		buffer[i] = "1234567890"[int(buffer[i])%6]
+	}
+	host_vcode := string(buffer)
+
+	// SendVeriMsg(Email, host_vcode) // Func wait for implementation
+
+	// Wait for response from client...
+
+	client_vcode := c.QueryParam("VeriCode")
+	new_passwd := c.QueryParam("Passwd")
+
+	if client_vcode == host_vcode {
+		return modifyPasswd(c, Email, new_passwd)
+	} else {
+		return c.JSON(http.StatusBadRequest, map[string]string{
+			"error": "Wrong Verification Code"})
+	}
+}
+
+// @Summary the interface of modifying password
+// @Description can only be called during logged-in status since there is no password check
+// @Tags Account
+// @Produce json
+// @Param email formData string true "user e-mail"
+// @Param paswd formData string true "user password (the new one)"
+// @Success 200 {string} string "{"msg": "Successfully modified"}"
+// @Failure 400 {string} string "{"error": "Invali E-mail Address"}"
+// @Router /case/PastHistory/{id} [PATCH]
+func (h *AccountHandler) ModifyPasswd(c echo.Context) error {
```

Comment:
You may need a general `ModifyUser` rather than limiting it to `ModifyPassword`

Reply:
s/changed/unchanged/ ?
That's reasonable.

Reply:
unchanged..